tests/functional/test_snap_dcgm.py

```diff
@@ -1,86 +1,138 @@
+        ), f"Failed to set {config} to {value}"
 
         # restart the service to apply the configuration
         subprocess.run(f"sudo snap restart {service}".split(), check=True)
 
+    @classmethod
+    def check_bind_config(cls, service: str, bind: str) -> None:
+        """Check if a service is listening on a specific bind."""
         for attempt in Retrying(wait=wait_fixed(2), stop=stop_after_delay(10)):
             with attempt:
                 assert 0 == subprocess.call(
-                    f"nc -z localhost {value.lstrip(':')}".split()
-                ), f"{service} is not listening on {value}"
+                    f"nc -z localhost {bind.lstrip(':')}".split()
+                ), f"{service} is not listening on {bind}"
+
+    @classmethod
+    def check_config_exists(cls, config: str) -> str:
+        """Check if a configuration exists in the snap configuration.
+
+        :return: The value of the current configuration
+        """
+        result = subprocess.check_output("sudo snap get dcgm -d".split(), text=True)
+        dcgm_snap_config = json.loads(result.strip())
+        assert config in dcgm_snap_config, f"{config} is not in the snap configuration"
+        return str(dcgm_snap_config[config])
+
+    @classmethod
+    @retry(wait=wait_fixed(2), stop=stop_after_delay(10))
+    def check_metric_config(cls, metric_file: str) -> None:
+        dcgm_exporter_service = "snap.dcgm.dcgm-exporter"
+
+        result = subprocess.check_output(
+            f"sudo systemctl show -p ActiveEnterTimestamp {dcgm_exporter_service}".split(),
+            text=True,
+        )
+
+        start_timestamp = result.strip().split("=")[1]
+
+        result = subprocess.check_output(
+            f"sudo journalctl -u {dcgm_exporter_service} --since '{start_timestamp}'",
+            shell=True,
+            text=True,
+        )
+
+        assert metric_file in result, f"Metric file {metric_file} is not loaded"
+
+    @pytest.mark.parametrize(
+        "service, config, new_value",
+        [
+            ("dcgm.dcgm-exporter", "dcgm-exporter-address", ":9466"),
+            ("dcgm.nv-hostengine", "nv-hostengine-port", "5566"),
+        ],
+    )
+    def test_dcgm_bind_config(self, service: str, config: str, new_value: str) -> None:
+        """Test snap bind configuration."""
+        old_value = self.check_config_exists(config)
+
+        self.set_config(service, config, new_value)
+        self.check_bind_config(service, new_value)
+
+        # Revert back
+        self.set_config(service, config, old_value)
+        self.check_bind_config(service, old_value)
+
+    def test_dcgm_metric_config(self) -> None:
+        service = "dcgm.dcgm-exporter"
+        config = "dcgm-exporter-metrics-file"
+        metric_file = "test-metrics.csv"
+        metric_file_path = os.path.join(os.getenv("SNAP_COMMON"), metric_file)
+
+        self.check_config_exists(config)
+
+        subprocess.check_call(f"sudo touch {metric_file_path}".split())
```

Comment:
The reason why I am using subprocess for creating a file is that you need sudo to create it in `$SNAP_COMMON`. If I use `open` or `pathlib` it will throw `PermissionError`. The only way to tackle this is to set different permissions on the directory before creating the file or running `tox` with sudo, which presents a different problem where the command will not be found, and you need to either install tox for root or provide a full path to the executable which will complicate current CI setup.

Reply:
makes sense. Please add a comment why you are using subprocess